src/impl_bytemuck.rs

```diff
@@ -0,0 +1,259 @@
+// Copyright 2024 the Peniko Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#![allow(unsafe_code, reason = "unsafe is required for bytemuck unsafe impls")]
+
+use crate::{Compose, Extend, Fill, Mix};
+
+// Safety: The enum is `repr(u8)` and has only fieldless variants.
+unsafe impl bytemuck::NoUninit for Compose {}
+
+// Safety: The enum is `repr(u8)` and `0` is a valid value.
+unsafe impl bytemuck::Zeroable for Compose {}
+
+// Safety: The enum is `repr(u8)`.
+unsafe impl bytemuck::checked::CheckedBitPattern for Compose {
+    type Bits = u8;
+
+    fn is_valid_bit_pattern(bits: &u8) -> bool {
+        use bytemuck::Contiguous;
+        // Don't need to compare against MIN_VALUE as this is u8 and 0 is the MIN_VALUE.
+        *bits <= Self::MAX_VALUE
+    }
+}
+
+// Safety: The enum is `repr(u8)`. All values are `u8` and fall within
+// the min and max values.
+unsafe impl bytemuck::Contiguous for Compose {
+    type Int = u8;
+    const MIN_VALUE: u8 = Self::Clear as u8;
```

Comment:
I wonder if we should validate that this is actually zero somewhere? I.e. `assert!(Compose::MIN_VALUE==0)`
(or a const-permitted equivalent) in the corresponding `const _: ()`

Reply:
Handled essentially by checking that `zeroed` is `Self::Clear` below ... and this is u8, so won't be anything lower.

Reply:
Grand. That works for me. Thanks